### PR TITLE
new Maker for api-platform data persisters

### PR DESCRIPTION
**Pull request by zairigimad**
### New Command to generate API DataPersister
I Like Symfony and API Platform, I would like to leverage the power of Maker Bundle to generate code for API Platform.

Makers for API Platform:
In this PR:
- [x] DataPersister

Next:
- [ ] DataProvider

- [ ] DataTransformer

This PR is for DataPersister Maker.

Capture d’écran 2020-10-24 à 22 11 52

generated code : 

<details>
  <summary>ArticleDataPersister.php</summary>
  
```php
<?php

namespace App\DataPersister;

use ApiPlatform\Core\DataPersister\ContextAwareDataPersisterInterface;

final class ArticleDataPersister implements ContextAwareDataPersisterInterface
{
    public function supports($data, array $context = []): bool
    {
        // implement your logic to check whether the given data is supported by this data persister
        return true;
    }

    public function persist($data, array $context = [])
    {
        // call your persistence layer to save $data
        return $data;
    }

    public function remove($data, array $context = [])
    {
        // call your persistence layer to delete $data
    }
}

```
</details>

### CODE REVIEW

**Review by weaverryan — COMMENTED** [review]
I would absolutely LOVE to have a whole series of API Platform generates. So big 💯 from me.

I've added some notes to take the `make:api:data-persister` to the next level :). And let's make just 1 PR per command so that we can merge each as soon as its done and stay focused.

Thanks!

File: src/Maker/MakeDataPersister.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            ->setHelp(file_get_contents(__DIR__.'/../Resources/help/MakeDataPersister.txt'));
+    }
+
+    public function generate(InputInterface $input, ConsoleStyle $io, Generator $generator)
+    {
+        $dataPersisterClassNameDetails = $generator->createClassNameDetails(
+            $input->getArgument('name'),
+            'DataPersister\\',
+            'DataPersister'
+        );
```
[the remark this answers is not in the document]
Comment (zairigimad) [reply src/Maker/MakeDataPersister.php:49]: yes, great ideas, I'm gonna Try to add this, and make Seperate PRs for the other Makers :)

[the remark this answers is not in the document]
Comment (zairigimad) [reply src/Maker/MakeDataPersister.php:111]: @weaverryan the Maker is now working with the update of services and with the different cases.
I would love to see your feedback if there are somethings to refactor in the code.

**Comment by weaverryan** [comment]
Hey @zairigimad!

I just talked with @dunglas and he was also thinking we should have these maker commands (and so I pointed him to your PR!). But we think it makes most sense for these to live inside of API Platform itself, so that once you install API Platform, you have the commands. Would you be willing to re-make this PR to API Platform? I'm not sure if it would just "drop in" or if some extra work would be needed - I'd be happy to help.

Thank you!

**Comment by weaverryan** [comment]
Ah, @dunglas just told me that a Maker has been started already on API Platform! https://github.com/api-platform/core/pull/3850

That is probably the PR we should use - it's clearly a good idea, as multiple people are proposing it (and I also think it's a good idea). So, unfortunately @zairigimad even though this is aa very good PR, I'm going to close it in favor of the other one on API Platform. I hope you could review that one over there and propose any improvements from your PR.

Thanks!

**Comment by zairigimad** [comment]
> Hey @zairigimad!
> 
> I just talked with @dunglas and he was also thinking we should have these maker commands (and so I pointed him to your PR!). But we think it makes most sense for these to live inside of API Platform itself, so that once you install API Platform, you have the commands. Would you be willing to re-make this PR to API Platform? I'm not sure if it would just "drop in" or if some extra work would be needed - I'd be happy to help.
> 
> Thank you!

Hey @weaverryan ,

Sure, no problem, I will check the PR in Api-Platform, and maybe will add other Makers :)

**Pull request closed without merging**